crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -721,15 +734,15 @@ impl Room {
             }
         }
 
-        Ok(names)
+        Ok((names, num_service_members))
     }
 
     /// Compute the room summary with the data present in the store.
     ///
     /// The summary might be incorrect if the database info is outdated.
     ///
     /// Returns a `(heroes_names, num_joined_invited)` tuple.
-    async fn compute_summary(&self) -> StoreResult<(Vec<String>, u64)> {
+    async fn compute_summary(&self) -> StoreResult<(Vec<String>, u64, u64)> {
```

Comment:
Same here, can you update the above doc comment? (or return a tiny struct here, since this is returning 3 values that would make sense to me)

Reply:
Something like this: https://github.com/matrix-org/matrix-rust-sdk/pull/4407/commits/bb0baf0b0d2f803d72c312de882ec319b1449169?

I made both paths return the struct, now I wonder if it would be simpler if the joined/invted count manipulation should just go into each subfunction as well.

Reply:
Yes (to both points) :+1: